core/amber/src/main/scala/edu/uci/ics/texera/web/auth/UserAuthenticator.scala

```diff
@@ -15,7 +15,7 @@ object UserAuthenticator extends Authenticator[JwtContext, SessionUser] with Laz
     try {
       val userName = context.getJwtClaims.getSubject
       val userId = UInteger.valueOf(context.getJwtClaims.getClaimValue("userId").asInstanceOf[Long])
-      val user = new User(userName, userId, null, null)
+      val user = new User(userName, userId, null, null, null)
       Optional.of(new SessionUser(user, Set(SessionRole.BASIC)))
```

Comment:
As discussed, we need to change the role to each user based on their permission (or role) saved in db. Then for endpoints, we need to update which one we permit admin only and which ones we permit user and admin, etc.

Reply:
We are not there yet. Here is just the adding one role for login. I will add admin-related roles in the next PR.

Reply:
yeah, but currently in this PR, how do we prevent `pending` users from accessing endpoints besides login? are you doing a frontend check?